core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -47,6 +47,14 @@ public static MatchAllQueryBuilder matchAllQuery() {
     }
 
     /**
+     * A query that match no documents.
+     */
+    // NO COMMIT: we should create a new MatchNoDocsQueryBuilder class for this instead?
+    public static EmptyQueryBuilder matchNoDocsQuery() {
+        return EmptyQueryBuilder.PROTOTYPE;
```

Comment:
I think IdsQueryBuilder() without specifying any id returns `Queries.newMatchNoDocsQuery()` lucene query. EmptyQueryBuilder returns `null` when `toQuery()` is called, so this won't work.